File: Python3/Majority Element.py
# 52ms, 100.0%
class Solution:
    def majorityElement(self, nums):
        cur, count = nums[0], 0
        for num in nums:
            if cur == num:
                count += 1
            else:
                count -= 1
                if count == 0: cur, count = num, 1
        return cur
        '''
        ### 52ms, 100.0%
        return sorted(nums)[len(nums) // 2]
        '''
        

File: Java/Majority Element.java
// 1ms, 99.87%; 41.8MB, 68.38%
class Solution {
    public int majorityElement(int[] nums) {
        int res = nums[0], count = 1;
        for (int i = 1; i < nums.length; ++i) {
            count += (nums[i] == res ? 1 : -1);
            if (count == 0) {
                res = nums[i];
                count = 1;
            }
        }
        return res;
    }
}
